internal/core/graphics/brush.rs

```diff
@@ -254,20 +260,163 @@ impl RadialGradientBrush {
+                let color_at_0 = Self::interpolate_color(below.color, above.color, t);
+                stops.insert(0, GradientStop { position: 0.0, color: color_at_0 });
+            } else if let Some(above) = stop_above_0 {
+                // Only stops above 0, use the first stop's color
+                stops.insert(0, GradientStop { position: 0.0, color: above.color });
+            }
+        }
+
+        // Add interpolated boundary stop at 1.0 if needed
+        let has_stop_at_1 = stops.iter().any(|s| (s.position - 1.0).abs() < f32::EPSILON);
+        if !has_stop_at_1 {
+            // For 1.0: find the stop just at/below 1 and just above 1
+            let stop_below_1 = stops.iter().filter(|s| s.position <= 1.0).last(); // closest to 1 from below
+            let stop_above_1 = stops.iter().filter(|s| s.position > 1.0).next(); // closest to 1 from above
+
+            if let (Some(below), Some(above)) = (stop_below_1, stop_above_1) {
+                // Interpolate between the stop below 1 and the stop above 1
+                // Example: 350deg and 370deg → interpolate at 360deg
+                let t = (1.0 - below.position) / (above.position - below.position);
+                let color_at_1 = Self::interpolate_color(below.color, above.color, t);
+                stops.push(GradientStop { position: 1.0, color: color_at_1 });
+            } else if let Some(below) = stop_below_1 {
+                // Only stops below 1, use the last stop's color
+                stops.push(GradientStop { position: 1.0, color: below.color });
+            }
+        }
+
+        // Drop stops under 0deg and over 360deg
+        stops = stops.into_iter().filter(|s| 0.0 <= s.position && s.position <= 1.0).collect();
+
+        // Adjust first stop (at 0.0) to avoid duplicate with stop at 1.0
+        if let Some(first) = stops.first_mut() {
+            if first.position.abs() < f32::EPSILON {
+                first.position = f32::EPSILON;
+            }
+        }
+
+        Self { from_angle, stops: SharedVector::from_iter(stops.into_iter()) }
     }
 
     /// Returns the color stops of the conic gradient.
     pub fn stops(&self) -> impl Iterator<Item = &GradientStop> {
-        self.0.iter()
+        self.stops.iter()
+    }
+
+    /// Returns the color stops with the `from_angle` rotation applied.
+    ///
+    /// This method returns a new vector of stops where:
+    /// 1. Stops outside [0.0, 1.0] range are removed
+    /// 2. Boundary stops at 0.0 and 1.0 are added if missing (with interpolated colors)
+    /// 3. Each stop's position is adjusted by adding `from_angle` and wrapping to [0.0, 1.0]
+    /// 4. The stops are sorted by their rotated positions
+    ///
+    /// This is useful when you need to work with the actual visual positions of the stops
+    /// after the gradient has been rotated.
+    pub fn rotated_stops(&self) -> alloc::vec::Vec<GradientStop> {
```

Comment:
Shouldn't we be storing them like this instead of re-allocating a new vector every time we want to draw?

Reply:
You are right. I updated `new` to do everything needed.